File: app/src/test/java/com/coinninja/coinkeeper/view/subviews/SharedMemoViewTest.java
package com.coinninja.coinkeeper.view.subviews;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.coinninja.android.helpers.Resources;
import com.coinninja.coinkeeper.R;
import com.coinninja.coinkeeper.TestCoinKeeperApplication;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import static com.coinninja.android.helpers.Views.withId;
import static com.coinninja.matchers.TextViewMatcher.hasText;
import static com.coinninja.matchers.ViewMatcher.hasTag;
import static org.hamcrest.MatcherAssert.assertThat;


@RunWith(RobolectricTestRunner.class)
@Config(application = TestCoinKeeperApplication.class)
public class SharedMemoViewTest {

    private View view;
    private Context context;

    @Before
    public void setUp() throws Exception {
        context = RuntimeEnvironment.application;
        view = LayoutInflater.from(context).inflate(R.layout.shared_memo_container, null);

    }

    @Test
    public void test_configure_for_sharing__renders_shared_icon() {
        SharedMemoView sharedMemoView = new SharedMemoView(view, true, "some memo", "Jon Doe");

        ImageView imageView = withId(view, R.id.shared_status_image_view);
        assertThat(imageView, hasTag(R.drawable.ic_shared_user));
    }

    @Test
    public void sets_memo_status_text() {
        SharedMemoView sharedMemoView = new SharedMemoView(view, true, "some memo", "Jon Doe");

        TextView textView = withId(view, R.id.shared_memo_status_text_view);

        assertThat(textView, hasText(Resources.getString(context, R.string.shared_memo, "Jon Doe")));
    }
}

File: app/src/test/java/com/coinninja/coinkeeper/view/subviews/SharedMemoViewTest.kt
package com.coinninja.coinkeeper.view.subviews

import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.test.core.app.ApplicationProvider
import androidx.test.ext.junit.runners.AndroidJUnit4
import com.coinninja.coinkeeper.R
import com.google.common.truth.Truth.assertThat
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class SharedMemoViewTest {

    private fun createView(): View = LayoutInflater.from(ApplicationProvider.getApplicationContext())
            .inflate(R.layout.shared_memo_container, null)

    @Test
    fun test_configure_for_sharing__renders_with_shared_icon() {
        val view = createView()
        SharedMemoView().render(view, true, "some memo", "Jon Doe")
        assertThat(view.findViewById<TextView>(R.id.shared_memo_status_text_view).text.toString()).isEqualTo("Will be seen by Jon Doe")
        assertThat(view.findViewById<TextView>(R.id.shared_memo_text_view).text.toString()).isEqualTo("some memo")
        assertThat(view.findViewById<ImageView>(R.id.shared_status_image_view).tag).isEqualTo(R.drawable.ic_shared_user)
        assertThat(view.visibility).isEqualTo(View.VISIBLE)
    }

    @Test
    fun test_configure_for_sharing__renders_with_out_shared_icon() {
        val view = createView()
        SharedMemoView().render(view, false, "some memo", "Jon Doe")
        assertThat(view.findViewById<TextView>(R.id.shared_memo_status_text_view).text.toString()).isEqualTo("Will be seen by only you.")
        assertThat(view.findViewById<TextView>(R.id.shared_memo_text_view).text.toString()).isEqualTo("some memo")
        assertThat(view.findViewById<ImageView>(R.id.shared_status_image_view).tag).isEqualTo(R.drawable.ic_single_user)
        assertThat(view.visibility).isEqualTo(View.VISIBLE)
    }

    @Test
    fun hidden_when_empty_memo() {
        val view = createView()
        SharedMemoView().render(view, false, "", "Jon Doe")
        assertThat(view.visibility).isEqualTo(View.GONE)
    }
}